heron/api/src/java/com/twitter/heron/api/Config.java

```diff
@@ -147,6 +147,31 @@
    */
   public static final String TOPOLOGY_COMPONENT_RAMMAP = "topology.component.rammap";
   /**
+   * Is this topology stateful? The format of this flag is boolean
+   */
+  public static final String TOPOLOGY_STATEFUL = "topology.stateful";
```

Comment:
Can we name this in a way that implies that heron state management is enabled or disabled by this boolean flag? It should be very clear to the user the effect of using `IStatefulComponents` for example with this set to `false`. Or what this should be set to for an externally stateful topology that doesn't use `IStatefulComponents`. As named it sounds like it describes a characteristic of the topology, versus the behavior of the system.